install/helm/gloo/templates/5-resource-configmap.yaml

```
@@ -12,3 +12,13 @@ data:
   custom-resources: |
 {{ $cr }}
   has-custom-resources: "{{- $hasCr -}}"
```

as an alternative, we can also stuff all the EE and OSS resources into the same `custom-resources` field, and then the rollout job / rollout check job just needs to read from the one field.

---

The enterprise upstreams should first be applied successfully since they need to exist before they are referenced in settings (eg. extauth). If all the resources are applied in a single go, the upstream might fail to be created and the setting which refers to that upstream will also fail.
This might be trivial since the job will retry but I'd like to preserve the ordering

---

agreed that we need to preserve the right ordering. but if we put them all in the same yaml in the correct order, kubectl should apply them in that order, right? i'm not sure though what happens if one of the resources fails to apply - does kubectl keep trying to apply the remaining resources, or does it fail as soon as one resource fails?

---

Running a quick test, kubectl apply continues to apply resources in the file even some of them are invalid but still returns an exit value of 1

```
apiVersion: gloo.solo.io/v1
kind: Upstream
...
---
apiVersion: gateway.solo.io/v1
kind: Gateway
...
```

```
gateway.gateway.solo.io/abc-proxy created
The Upstream "rateLimit" is invalid: metadata.name: Invalid value: "rateLimit": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')

```

---

ok in that case, if we want to fail early, i think it's fine to split custom-resources and extra-custom-resources as you have.
but can we indicate in the error message (see other comment below) something like "there may also be other resources (e.g. gateways) that were not applied" so they know it's not just the upstreams that failed to apply?

---

Good idea. Added the changes